### PR TITLE
125 add price to order items table

### LINKED ISSUE #111: Add Orders to seed

### LINKED ISSUE #122: Add price to order_item table
- so if item price changes the order price does not change.

### LINKED ISSUE #125: Admin items index
- change from cards to collection

### PR DESCRIPTION
**Pull request by meyerhoferc**
* adds price for order-item table
* fixes many views for consistency
* adds item show view for admin
* changes admin item index to striped collection table

all tests pass
closes #111 
closes #125 
closes #122

### CODE REVIEW

File: app/controllers/admin/items_controller.rb [hunk 1]
```diff
@@ -36,6 +36,10 @@ def update
     end
   end
 
+  def show
+    @item = Item.find(params[:id])
+  end
+
```
Comment (cmacaulay) [inline app/controllers/admin/items_controller.rb:42]: 🎉

File: app/models/order.rb [hunk 2]
```diff
@@ -21,7 +21,8 @@ def self.get_by_status(find_status)
   def submit_order(cart)
     cart.items.each do |item, quantity|
       order_items.create!(item_id: item.id,
-                          quantity: quantity)
+                          quantity: quantity,
+                          price: item.price)
     end
```
Comment (cmacaulay) [inline app/models/order.rb:26]: 👍 Looks great!

File: app/models/order_item.rb [hunk 3]
```diff
@@ -2,7 +2,7 @@ class OrderItem < ApplicationRecord
   belongs_to :order
   belongs_to :item
 
-  validates :quantity, presence: true
+  validates :quantity, :price, presence: true
```
Comment (cmacaulay) [inline app/models/order_item.rb:5]: Great! Love that you added this to the model test as well.

File: app/views/items/show.html.erb [hunk 4]
```diff
@@ -8,14 +7,18 @@
...
+          <img class="image-format" src="<%= @item.image.url %>"/>
+        <% end %>
       </div>
       <div class="card-content">
           <span class="card-title"><%= @item.title %></span>
           <ul>
             <li>Price: <%= number_to_currency(@item.price, precision: 2) %></li>
             <li>Category: <%= @item.category.name %></li>
-            <li>Description: <%= @item.description %></li>
+            <li class="truncate">Description: <%= @item.description %></li>
```
Comment (cmacaulay) [inline app/views/items/show.html.erb:21]: 👍 thanks, materialize

File: app/views/shared/_items.html.erb [hunk 5]
```diff
@@ -3,7 +3,11 @@
   <div class="col s4">
     <div class="card">
       <div class="card-image">
-        <img class="image-format" src="<%= item.image_tag %>" />
+        <% if item.image_tag %>
+          <img class="image-format" src="<%= item.image_tag %>"/>
+        <% else %>
+          <img class="image-format" src="<%= item.image.url %>"/>
+        <% end %>
```
Comment (cmacaulay) [inline app/views/shared/_items.html.erb:10]: Hopefully this logic goes away after paperclip is up and running.

File: db/schema.rb [hunk 6]
```diff
@@ -43,6 +43,7 @@
     t.integer  "quantity"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
+    t.float    "price"
     t.index ["item_id"], name: "index_order_items_on_item_id", using: :btree
     t.index ["order_id"], name: "index_order_items_on_order_id", using: :btree
   end
```
Comment (cmacaulay) [inline db/schema.rb:49]: 🎉

File: db/seeds.rb [hunk 7]
```diff
@@ -13,24 +13,47 @@ def import_product_csv
...
+order2 = user.orders.create!(total_price: (item3.price + item4.price + item5.price).round(2))
+order2.order_items.create(item: item3, quantity: 1, price: item3.price)
+order2.order_items.create(item: item4, quantity: 1, price: item4.price)
+order2.order_items.create(item: item5, quantity: 1, price: item5.price)
+puts "Order 2 added!"
+
+order3 = user.orders.create!(total_price: (item3.price * 2 + item5.price).round(2), status: "canceled")
+order3.order_items.create(item: item3, quantity: 2, price: item3.price)
+order3.order_items.create(item: item5, quantity: 1, price: item5.price)
+puts "Order 3 added!"
```
Comment (cmacaulay) [inline db/seeds.rb:59]: Great seed, wondering if there's a way to refactor this - potentially an enumerable through order.items? Not high priority, curious about seeding best practices.

File: spec/fabricators/order_item_fabricator.rb [hunk 8]
```diff
@@ -2,4 +2,5 @@
   order    Fabricate(:order)
   item     Fabricate(:item)
   quantity 1
+  price    item.price
```
Comment (cmacaulay) [inline spec/fabricators/order_item_fabricator.rb:5]: 👍

File: spec/features/admin/items/admin_can_view_all_items_spec.rb [hunk 9]
```diff
@@ -6,34 +6,40 @@
...
         expect(page).to have_link("Edit")
       end
 
-      within(cards[2]) do
-        expect(page).to have_content(item3.title)
+      within(items[3]) do
+        expect(page).to have_link(item3.title)
         expect(page).to have_content(item3.description)
         expect(page).to have_content(item3.item_status)
         expect(page).to have_link("Edit")
```
Comment (cmacaulay) [inline spec/features/admin/items/admin_can_view_all_items_spec.rb:45]: Great test!

**Review by cmacaulay — APPROVED** [review]
Looks good! I removed a save_and_open_page.

File: app/views/orders/show.html.erb [hunk 10]
```diff
@@ -11,18 +11,18 @@
...
     </thead>
     <tbody>
-      <% @order.items.each do |item| %>
+      <% @order.order_items.each do |order_item| %>
       <tr>
-        <td><%= link_to item.title, item_path(item) %></td>
-        <td><%= number_to_currency(item.price, precision: 2) %></td>
-        <td><%= item.get_quantity_in_order(@order)%></td>
-        <td><%= number_to_currency(item.get_subtotal_on_order(@order), precision: 2) %></td>
+        <td><%= link_to order_item.item.title, item_path(order_item.item) %></td>
```
Comment (CjMoore) [inline app/views/orders/show.html.erb:16]: I like this. We should look into if the methods .get_subtotal_on_item and .get_quantity_in_order are used anywhere else, if not we can take them out.

**Review by CjMoore — APPROVED** [review]

**Pull request merged**